x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/search/SearchApplication.java

```diff
@@ -90,7 +93,15 @@ public SearchApplication(
 
     public SearchApplication(StreamInput in) throws IOException {
         this.name = in.readString();
-        this.indices = in.readStringArray();
+        this.indices = null;
+        this.analyticsCollectionName = in.readOptionalString();
+        this.updatedAtMillis = in.readLong();
+        this.searchApplicationTemplate = in.readOptionalWriteable(SearchApplicationTemplate::new);
+    }
+
+    public SearchApplication(StreamInput in, String[] indices) throws IOException {
+        this.name = in.readString();
+        this.indices = indices;
```

Comment:
I'd define a new constant:

```java
private static final TransportVersion INDICES_REMOVED_TRANSPORT_VERSION = TransportVersion.V_8_500_069;
```

and change this to check which transport version we are using:

```suggestion
if (in.getTransportVersion().onOrAfter(INDICES_REMOVED_TRANSPORT_VERSION) {
        this.indices = indices; // Uses the provided indices, as they are no longer serialized
} else {
        this.indices = in.readStringArray(); // old behaviour, read it from input as it was serialized
}
```